tendermint-rs/src/secret_connection/mod.rs

```diff
@@ -123,34 +111,28 @@ impl<IoHandler: Read + Write + Send + Sync> SecretConnection<IoHandler> {
     }
 
     /// Unseal (i.e. decrypt) AEAD authenticated data
-    fn open(&self, authtext: &[u8], ciphertext: &[u8], out: &mut [u8]) -> Result<usize, Error> {
+    fn open(&self, ciphertext: &[u8], out: &mut [u8]) -> Result<usize, Error> {
+        let nonce = aead::Nonce::from(&self.recv_nonce);
+        let associated_data = aead::Aad::empty();
+
         // optimize if the provided buffer is sufficiently large
-        if out.len() >= ciphertext.len() {
+        let len = if out.len() >= ciphertext.len() {
             let in_out = &mut out[..ciphertext.len()];
             in_out.copy_from_slice(ciphertext);
-            let len = aead::open_in_place(
-                &self.recv_secret,
-                &self.recv_nonce.to_bytes(),
-                authtext,
-                0,
-                in_out,
-            )
-            .map_err(|_| Error::Crypto)?
-            .len();
-            Ok(len)
+
+            aead::open_in_place(&self.recv_secret, nonce, associated_data, 0, in_out)
+                .map_err(|_| Error::Crypto)?
+                .len()
         } else {
             let mut in_out = ciphertext.to_vec();
-            let out0 = aead::open_in_place(
-                &self.recv_secret,
-                &self.recv_nonce.to_bytes(),
-                authtext,
-                0,
-                &mut in_out,
-            )
-            .map_err(|_| Error::Crypto)?;
+            let out0 =
+                aead::open_in_place(&self.recv_secret, nonce, aead::Aad::empty(), 0, &mut in_out)
```

Comment:
Couldn't this also be:
```suggestion
                aead::open_in_place(&self.recv_secret, nonce, associated_data, 0, &mut in_out)
```
Or, delete `let associated_data = aead::Aad::empty();` above and use `aead::Aad::empty()` in both cases?